Weave IA refresh (#2028)
wandb/docs: weave/concepts/what-is-weave.mdx

---
title: "What is Weave?"

> description: "Learn what W&B Weave is and how it helps you build, evaluate, and improve LLM applications"

Reviewer: ```suggestion
description: "Learn about W&B Weave and how it helps you build, evaluate, and improve LLM applications"
```

---

W&B Weave is an observability and evaluation platform for building reliable LLM applications. Weave helps you understand what your AI application is doing, measure how well it performs, and systematically improve it over time.

> ## Why Weave?

Reviewer: I don't love questions in headings and this seems strangely placed. Maybe it can be omitted.

Building LLM applications is fundamentally different from traditional software development. LLM outputs are non-deterministic, making debugging harder. Quality is subjective and context-dependent. Small prompt changes can cause unexpected behavior changes. Traditional testing approaches fall short.

Weave addresses these challenges by providing:

- **Visibility** into every LLM call, input, and output in your application
- **Systematic evaluation** to measure performance against curated test cases
- **Version tracking** for prompts, models, and data so you can understand what changed
- **Feedback collection** to capture human judgments and production signals

> ## What you can do with Weave

Reviewer: This could be omitted and the H3s could be promoted to H2.

Author: ah, yeah. I totally deleted a description from under there and forgot to kill the header

### Evaluate systematically

Run your application against curated test datasets and measure performance with scoring functions. Track how changes to prompts or models affect quality over time.

[Build an evaluation pipeline](/weave/tutorial-eval)

### Version everything

Weave tracks versions of your prompts, datasets, and model configurations. When something breaks, you can see exactly what changed. When something works, you can reproduce it.

[Learn about versioning](/weave/guides/tracking/objects)

## Experiment with prompts and models

Bring your API keys and quickly test prompts and compare responses from various commercial models using the Playground.

[Experiment in the Weave Playground](/weave/guides/tools/playground)

### Collect feedback

Capture human feedback, annotations, and corrections from production use. Use this data to build better test cases and improve your application.

[Collect feedback](/weave/guides/tracking/feedback)

### Monitor production

Score production traffic with the same scorers you use in evaluation. Set up guardrails to catch issues before they reach users.

[Set up guardrails and monitors](/weave/guides/evaluation/guardrails_and_monitors)

> ## How Weave fits in your workflow

Reviewer: Not sure about this heading either, the contents of the section kind of imply "everywhere".

Author: you know what? I hate this section and I don't think it's necessary. Nixing it

Weave supports the full LLM application development lifecycle:

| Phase | What Weave provides |
|-------|---------------------|
| **Build** | Trace calls to understand behavior, debug issues, and iterate quickly |
| **Test** | Evaluate against datasets with custom and built-in scorers |
| **Deploy** | Version prompts and models for reproducible deployments |
| **Monitor** | Score production traffic, collect feedback, catch regressions |

## Supported languages

Weave provides SDKs for Python and TypeScript:

<Tabs>

>   <Tab title="Python">

Reviewer: I've seen problems with trying to indent within tabsets in Mintlify, FWIW

```bash
pip install weave
```
  </Tab>
  <Tab title="TypeScript">
```bash
npm install weave
```
  </Tab>
</Tabs>

> Both SDKs support tracing, evaluation, datasets, and the core Weave features. Some advanced features like class-based Models and Scorers are currently Python-only.

Reviewer: Similar comment as above where this might be more clear to just say that TypeScript doesn't support everything.
